includes/razorpay-webhook.php

```diff
@@ -91,7 +110,22 @@ protected function paymentAuthorized($data)
 
         $razorpayPaymentId = $data['payload']['payment']['entity']['id'];
 
-        $payment = $this->api->payment->fetch($razorpayPaymentId);
+        try
+        {
+            $payment = $this->api->payment->fetch($razorpayPaymentId);
+        }
+        catch (Exception $e)
+        {
+            $log = array(
+                'message'   => $e->getMessage(),
+                'data'      => $razorpayPaymentId,
+                'event'     => $data['event']
+            );
+
+            write_log($log);
+
+            exit;
```

Comment:
@StSarc if we do an exit, other hooks can't run. Since it was a razorpay webhook and couldn't be authenticated we don't want any further processing in this request.